test/expressions/test_expression_testing_framework.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+import subprocess
+import sys
+
+error_template = """
+Expected "{}" in stdout!
+
+actual stdout:
+{}
+
+
+actual stderr:
+{}
+"""
+
+
+class TestExpressionTestingFramework(unittest.TestCase):
+    def runCommand(self, command):
+        p1 = subprocess.Popen(command, stdout = subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+        out, err = p1.communicate()
```

Comment:
Huh I read more about communicate and I think it is the right thing. Dunno why this was hanging for me in Stan. Seems to work for me now.

Reply:
I dont exactly remember what I changed, but the hanging was because python in the subprocess was running in interactive mode instead of running the script.